### PR TITLE
Add Suilend Fees Adapter

### PR DESCRIPTION
**Pull request by DaSichuan**

### CODE REVIEW

**Comment by llamatester** [comment]
The suilend adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts fees suilend
    
    🦙 Running SUILEND adapter 🦙
    ---------------------------------------------------
    Start Date:	Tue, 22 Oct 2024 00:00:00 GMT
    End Date:	Wed, 23 Oct 2024 00:00:00 GMT
    ---------------------------------------------------
    
    SUI 👇
    Backfill start time: 1/3/2024
    End timestamp: 1729641599 (2024-10-22T23:59:59.000Z)
    Daily fees: 33.82 k
    └─ Methodology: Interest and fees paid by borrowers and the liquidated
    Daily user fees: 33.82 k
    Daily revenue: 16.55 k
    Daily protocol revenue: 16.55 k

File: fees/suilend/index.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  next: string;
+}
+
+const methodology = {
+  Fees: 'Interest and fees paid by borrowers and the liquidated',
+  ProtocolReveneue: 'The portion of the total fees going to the Suilend treasury'
+}
+
+const fetchSuilendStats = async (timestamp: number): Promise<FetchResultFees> => {
+  const url = `${suilendFeesURL}?ts=${timestamp}`
```
Comment (g1nt0ki) [inline fees/suilend/index.ts]: this timestamp is end of 24 hour period or start of?

**Review by g1nt0ki — CHANGES_REQUESTED** [review]
.

File: fees/suilend/index.ts (hunk 1 above)
Comment (DaSichuan) [reply fees/suilend/index.ts]: For whatever timestamp is passed in, it returns fees that fall within the 24 UTC date boundaries that the passed in timestamp is contained in. Same behavior as our Solend adapter

**Comment by llamatester** [comment]
The solend.ts adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts fees solend.ts
    
    🦙 Running SOLEND.TS adapter 🦙
    ---------------------------------------------------
    Start Date:	Tue, 29 Oct 2024 04:57:08 GMT
    End Date:	Wed, 30 Oct 2024 04:57:08 GMT
    ---------------------------------------------------
    
    SOLANA 👇
    Backfill start time: 30/1/2023
    Daily fees: 29.65 k
    └─ Methodology: Interest and fees paid by borrowers and the liquidated
    Daily user fees: 29.65 k
    Daily revenue: 11.01 k
    Daily protocol revenue: 11.01 k
    Daily supply side revenue: 18.09 k
    End timestamp: 1730264227 (2024-10-30T04:57:07.000Z)

**Comment by llamatester** [comment]
The suilend adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts fees suilend
    
    🦙 Running SUILEND adapter 🦙
    ---------------------------------------------------
    Start Date:	Tue, 29 Oct 2024 04:57:13 GMT
    End Date:	Wed, 30 Oct 2024 04:57:13 GMT
    ---------------------------------------------------
    
    SUI 👇
    Backfill start time: 1/3/2024
    Daily fees: 7.92 k
    └─ Methodology: Interest and fees paid by borrowers and the liquidated
    Daily user fees: 7.92 k
    Daily revenue: 5.00 k
    Daily protocol revenue: 5.00 k
    End timestamp: 1730264232 (2024-10-30T04:57:12.000Z)

**Pull request merged**